core/app/models/mno_enterprise/user.rb

```diff
@@ -71,143 +55,85 @@ class User < BaseResource
       validates :password, format: { with: Devise.password_regex, message: Devise.password_regex_message }, if: :password_required?
     end
 
-    #================================
-    # Associations
-    #================================
-    has_many :organizations, class_name: 'MnoEnterprise::Organization'
-    has_many :org_invites, class_name: 'MnoEnterprise::OrgInvite'
-    has_one :deletion_request, class_name: 'MnoEnterprise::DeletionRequest'
-    has_many :teams, class_name: 'MnoEnterprise::Team'
 
-    # Impac
-    has_many :dashboards, class_name: 'MnoEnterprise::Impac::Dashboard'
-    has_many :alerts, class_name: 'MnoEnterprise::Impac::Alert'
+    def initialize(params = {})
+     attributes
+      super
+    end
 
-    #================================
-    # Callbacks
-    #================================
-    before_save :expire_user_cache
+    custom_endpoint :create_api_credentials,  on: :member, request_method: :patch
+    custom_endpoint :authenticate, on: :collection, request_method: :post
 
     #================================
     # Class Methods
     #================================
     # The auth_hash includes an email and password
     # Return nil in case of failure
-    def self.authenticate(auth_hash)
-      u = self.post("user_sessions", auth_hash)
-
-      if u && u.id
-        u.clear_attribute_changes!
-        return u
+    def self.authenticate_user(auth_hash)
+      result = self.authenticate({data: {attributes: auth_hash}})
+      if result
```

Comment:
If we don't need the `clear_attribute_changes`:
```ruby
result = self.authenticate({data: {attributes: auth_hash}})
if (u = result&.first) && u.id
  u
end
```
no need for the nil at the end